### PR TITLE
more options around merging for protected branches

### PR DESCRIPTION
**Pull request by evansde77**
WIP for testing release merge with protected branches

### CODE REVIEW

File: src/cirrus/github_tools.py [hunk 1]
```diff
@@ -71,6 +73,60 @@ def branch_state(self, branch=None):
...
+        for x in self.branch_status_list(branch):
+            update_time = arrow.get(x['updated_at'])
+            delta = arrow.utcnow() - update_time
+            LOGGER.info("Branch status for {branch} last update: {time}".format(
+                branch=branch,
+                time=update_time.humanize()
+                ))
+            if delta.seconds > expire_time:
+                self.set_branch_state('success', x['context'], branch)
+
```
Comment (appeltel) [inline src/cirrus/github_tools.py]: Is this function supposed to be used somewhere?

Comment (evansde77) [inline src/cirrus/github_tools.py]: It was, but it doesnt actually clear the status, so will likely take it out in the final cut.

**Comment by evansde77** [comment]
un-WIP

**Comment by evansde77** [comment]
@cloudant/cloudant-sapi 

ready for review

**Comment by evansde77** [comment]
@appeltel @ksnavely @petevg @shudgston @jcounts

**Pull request merged**